sdk/daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/crypto/Hash.scala

```diff
@@ -180,18 +212,248 @@ object Hash {
       val ss = set.toSeq.sorted[String]
       iterateOver(ss.iterator, ss.size)(_ addString _)
     }
+
+    final def addOptional[S](opt: Option[S], hashS: this.type => S => this.type): this.type = {
+      opt match {
+        case None => addInt(0)
+        case Some(value) => hashS(addInt(1))(value)
```

Comment:
How about using `addByte` instead of `addInt`? Saves a couple of bytes.

Reply:
We are reusing the hash for Value that is used for authenticated contract Id. 
This is a bug we have to keep for backward compatibility. 
should we rewrite a full new Hashing scheme ?

Reply:
> We are reusing the hash for Value that is used for authenticated contract Id.

I thought that we had switched everything over to the upgrade-friendly hashing scheme by now and that one should use byte, not int. Am I mistaken here?

Reply:
From what @remyhaemmerle-da said the upgrade-friendly hashing scheme is not ready and we're better off using the previous one for 2.1

Reply:
Yes.
- the upgrade friendly is still not stable. 
- It seems a bit complicate to specify and be used by the external client
- Maybe you can make the tags for option parametric .